itests/utilities/common_tx_utility.go

```diff
@@ -373,6 +379,145 @@ func WaitForHeight(suite *f.BaseSuite, height uint64) uint64 {
+	require.NoError(suite.T(), err, "Couldn't get feature status info")
+	suite.T().Logf("Scala: Status of feature %d on height @%d: %s\n", featureID, h, status)
+	return status
+}
+
+func GetFeatureActivationHeightGo(suite *f.BaseSuite, featureID int, height uint64) int32 {
+	activationHeight, err := getFeatureActivationHeight(GetActivationFeaturesStatusInfoGo(suite, height), featureID)
+	require.NoError(suite.T(), err)
+	return activationHeight
+}
+
+func GetFeatureActivationHeightScala(suite *f.BaseSuite, featureID int, height uint64) int32 {
+	activationHeight, err := getFeatureActivationHeight(GetActivationFeaturesStatusInfoScala(suite, height), featureID)
+	require.NoError(suite.T(), err)
+	return activationHeight
+}
+
+func GetFeatureActivationHeight(suite *f.BaseSuite, featureID int, height uint64) int32 {
+	var err error
+	var activationHeight int32
+	activationHeight = -1
+	activationHeightGo := GetFeatureActivationHeightGo(suite, featureID, height)
+	activationHeightScala := GetFeatureActivationHeightScala(suite, featureID, height)
+	if activationHeightGo == activationHeightScala && activationHeightGo > -1 {
+		activationHeight = activationHeightGo
+	} else {
+		err = errors.New("Activation Height from Go and Scala is different")
+	}
+	require.NoError(suite.T(), err)
+	return activationHeight
+}
+
+func GetFeatureBlockchainStatus(suite *f.BaseSuite, featureID int, height uint64) (string, error) {
+	var status string
+	var err error
+	statusGo := GetFeatureBlockchainStatusGo(suite, featureID, height)
+	statusScala := GetFeatureBlockchainStatusScala(suite, featureID, height)
+	if statusGo == statusScala {
+		status = statusGo
+	} else {
+		err = errors.Errorf("Feature with Id %d has different statuses", featureID)
+	}
+	return status, err
+}
+
+func GetWaitingBlocks(suite *f.BaseSuite, height uint64, featureID int) uint64 {
+	var waitingBlocks uint64
+	votingPeriod := suite.Cfg.BlockchainSettings.ActivationWindowSize(height)
+	status, err := GetFeatureBlockchainStatus(suite, featureID, height)
+	require.NoError(suite.T(), err)
+	switch status {
+	case "ACTIVATED":
+		waitingBlocks = 0
+	case "APPROVED":
+		waitingBlocks = votingPeriod - (height - (height/votingPeriod)*votingPeriod)
+	case "UNDEFINED":
+		waitingBlocks = 2*votingPeriod - (height - (height/votingPeriod)*votingPeriod)
+	}
+	return waitingBlocks
+}
+
+func WaitForFeatureActivation(suite *f.BaseSuite, featureID int, height uint64) int32 {
+	var activationHeight int32
+	var err error
+	waitingBlocks := GetWaitingBlocks(suite, height, featureID)
+	h := WaitForHeight(suite, height+waitingBlocks)
+	activationHeightGo := GetFeatureActivationHeightGo(suite, featureID, h)
+	activationHeightScala := GetFeatureActivationHeightScala(suite, featureID, h)
+	if activationHeightScala == activationHeightGo {
+		activationHeight = activationHeightGo
+	} else {
+		err = errors.Errorf("Feature with Id %d has different activation heights", featureID)
```

Comment:
`suite.FailNowf` can be used here instead of `require` below.

Reply:
Done